src/librbd/io/ImageRequest.cc

```diff
@@ -826,6 +834,87 @@ void ImageWriteSameRequest<I>::update_stats(size_t length) {
   image_ctx.perfcounter->inc(l_librbd_ws_bytes, length);
 }
 
+template <typename I>
+uint64_t ImageCompareAndWriteRequest<I>::append_journal_event(
+    const ObjectRequests &requests, bool synchronous) {
+
+  I &image_ctx = this->m_image_ctx;
+
+  uint64_t tid = 0;
+  uint64_t buffer_offset = 0;
+  assert(!this->m_image_extents.empty());
```

Comment:
Nit: really shouldn't be possible to have more than a single image extent, so perhaps change to ```this->m_image_exents.size() == 1```